game-app/game-core/src/test/java/games/strategy/triplea/ui/UnitChooserTest.java

```diff
@@ -0,0 +1,68 @@
+package games.strategy.triplea.ui;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import games.strategy.engine.ClientFileSystemHelper;
+import games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel;
+import games.strategy.triplea.EngineImageLoader;
+import org.junit.jupiter.api.Test;
+import util.UiTest;
+
+class UnitChooserTest {
+    @Test
+    void getUiContextDecoration() {
+      getUiContextDecoration("big_world\\map\\games\\Big_World_1942_v3rules.xml");
+      getUiContextDecoration("270bc_wars\\map\\games\\270BC_Wars.xml");
+      getUiContextDecoration("conquest_of_the_world\\map\\games\\conquest_world.xml");
+    }
+
+  private void getUiContextDecoration(final String gameFile) {
+    final var gsm = new GameSelectorModel();
+    UiTest.setup(new String[0]);
```

Comment:
There are a number of topics to talk through together with `UiTest`.

Before we do that, it's worth considering if we should remove the `UiTest` for now and submit it as a follow-up PR. WDYT @RaiNova ?

I suggest this to:
- help get this overall work merged sooner
- reduce the number of open topics in this PR
- there are potentially several discussion threads that will need to play out, those discussions are perhaps better in a more focused context
- there are a number of details to review here (both high level and low). I'm concerned the PR & CR could be hijacked by that, and it could perhaps start to feel for both of us as very nitpicky.

Reply:
I agree. I am not so adept at configuration management. Don't grudge if the next commit is a bit strange regarding UnitChooserTest and UiTest. Are you okay accepting ```setCodeSourceFolder``` for the moment? I fear I get version management issues otherwise.

Reply:
Sure  :+1: